File: java17/src/test/java/org/chelck/dp/TimingsTest.java
package org.chelck.dp;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class TimingsTest {

     @Test
     public void test() throws Exception {
         Timings t = new Timings();
         t.start();
         Thread.sleep(10);
         t.mark();
         assertTrue(t.toString().contains("total:"));
         assertTrue(t.toString().contains("max:"));
     }

}

File: scala3_mutex/src/test/scala/org/chelck/dp/TimingsTest.scala
package org.chelck.dp

import org.scalatest.funsuite.AnyFunSuite
import org.scalatest.Assertions._

class TimingsTest extends AnyFunSuite {

  test("Timings has total and max data") {
    val t = new Timings(0L, 0L, 0L)
    t.start()
    Thread.sleep(10)
    t.mark()
    assert(t.toString.contains("total="))
    assert(t.toString.contains("max="))
  }
}
